build.js

```diff
@@ -16,81 +16,168 @@ function getDirname () {
 
 // it's basically the old file with /source/ replaced with /build/
 function getBuildFilePathFromSourceFilePath (sourceFilePath) {
-	// note that replace only replaces the first occurence if the first argument
+	// note that replace only replaces the first occurrence if the first argument
 	//  is a string
 	return sourceFilePath.replace(
 		getDirname() + '/source/',
 		getDirname() + '/build/'
 	)
 }
 
+function getFileExtension (filePath) {
+	return filePath.split('.').pop()
+}
+
 function isHtmlFile (filePath) {
-	return filePath.slice(-5) === '.html'
+	return getFileExtension(filePath) === 'html'
 }
 
-// create the target build/ folder if it doesn't exist yet
-if (!fs.existsSync('./build/')){
-	fs.mkdirSync('./build/')
-	console.log('Created build folder')
+function isCssFile (filePath) {
+	return getFileExtension(filePath) === 'css'
+}
+
+function makeFolder (folderPath) {
+	// console.log('Preparing folder', folderPath)
+	// create if it doesn't exists yet
+	if (!fs.existsSync(folderPath)) {
+		fs.mkdirSync(folderPath)
+		console.log('Created folder', folderPath)
+	}
+}
+
+// creates a dictionary of (filename: sourceFile) for Handlebars
+function makeComponentDictionary(globPattern) {
+	console.log('Looking for components matching: ' + globPattern)
+
+	const componentSourceFiles = glob.sync(globPattern)
+	const components = {}
+	for (const componentSourceFile of componentSourceFiles) {
+		const componentName = componentSourceFile
+			.split('/').pop() // get the actual filename
+			.replace(/\.html$/, '') // remove .html at the end
+	
+		console.log('Found component', componentName)
+
+		// parse component
+		const fileString = fs.readFileSync(componentSourceFile, "utf8")
+		if (fileString.indexOf('TEMPORARY_OPENING_SIGNATURE ') == 0) {
+			console.log(componentName, 'was identified as a compound component')
+
+			// split by sections
+			let sections = fileString.split('\r\nTEMPORARY_OPENING_SIGNATURE ')
+			sections[0] = sections[0].replace('TEMPORARY_OPENING_SIGNATURE ', '')
+			// seperate sections into name and content
+			sections = sections.map(section => section.split(' TEMPORARY_CLOSING_SIGNATURE\r\n'))
+			// add component sections to dictionary
+			let componentDict = {}
+			sections.map(section => componentDict[section[0]] = new Handlebars.SafeString(section[1]))
```

Comment:
map returns an array of outputs. In this case we throw away what is returned, so `forEach` would be more suitable

Reply:
Noted.
```
sections.forEach(
	section => 
	componentDict[section[0]] = new Handlebars.SafeString(section[1])
	)
```